### PR TITLE
feat: include node cache in keep-alive context

### PR DESCRIPTION
**Pull request by haakonflatval-cognite**
#### Type of change

Feat  

#### Jira ticket :blue_book:

https://cognitedata.atlassian.net/browse/UX-1456

## Description :pencil:

Make sure the `FdmNodeCache` is kept alive in the `RevealKeepAlive` context, alongside the Reveal viewer itself. This makes clicks responsive across the session, and it does not need to load the model asset mappings multiple times.

## How has this been tested? :mag:

With FDX

## Test instructions :information_source:

- Build branch and connect to FDX through e.g. `yalc`
- Start FDX
- Open 3D view of model with mapped nodes
- Keep the network tab open as you click on different mapped nodes
- Notice that the first click on a node spawns multiple requests, subsequent ones spawn none (except for the `graphql` query from the node preview card).
- Navigate away from the 3D view, e.g. by clicking `Open` on the preview card for a node
- Then navigate back again, e.g. by clicking the 3D-view icon in the top right
- Click some mapped nodes once again
- The clicks do not spawn new requests, as they would without this change

## Checklist :ballot_box_with_check:

- [x] I am proud of this feature.
- [x] I have performed a self-review of my own code.
- [x] I have added PR type (Feat, Bug, Chore, Test, Docs, Style, Refactor) to the PR title.
- [ ] I have manually tested this for different scenarios (different models, formats, devices, browsers).
- [ ] I have commented my code in hard-to-understand areas.
- [ ] I have made corresponding changes to the public documentation.
- [ ] I have added unit and visuals tests to prove that my feature works to the best of my ability.
- [ ] I have refactored the code for readability to the best of my ability.
- [ ] I have checked that my changes do not introduce regressions in the public documentation.
- [ ] I have outlined any known defects / lacking capabilities in the contents of this PR.
- [ ] I have listed any remaining work that should follow this PR in the description or jira/miro/etc.
- [ ] I have added TSDoc to any public facing changes.
- [ ] I have added "developer documentation" in any package README.md that is related / applicable to this PR.
- [ ] I have noted down and am currently tracking any technical debt introduced in this PR.
- [ ] I have thoroughly thought about the architecture of this implementation.
- [ ] I have asked peers to test this feature.

### CODE REVIEW

File: react-components/src/components/NodeCacheProvider/NodeCacheProvider.tsx [hunk 1]
```diff
@@ -97,8 +98,15 @@ export const useFdmAssetMappings = (
...
   const cdfClient = useSDK();
+  const revealKeepAliveData = useRevealKeepAlive();
 
-  const fdmCache = useMemo(() => new FdmNodeCache(cdfClient, fdmClient), []);
+  const fdmCache = useMemo(() => {
+    const cache = revealKeepAliveData?.fdmNodeCache.current ?? new FdmNodeCache(cdfClient, fdmClient);
+    if (revealKeepAliveData !== undefined) {
+      revealKeepAliveData.fdmNodeCache.current = cache;
+    }
+    return cache;
```
Comment (greenbech) [inline react-components/src/components/NodeCacheProvider/NodeCacheProvider.tsx]: I think this part is pretty hard to read. Can it be simplified to something like this instead?

```typescript
if (revealKeepAliveData.fdmNodeCache.current === undefined) {
  revealKeepAliveData.fdmNodeCache.current = new FdmNodeCache(..)
}
```

Comment (greenbech) [reply react-components/src/components/NodeCacheProvider/NodeCacheProvider.tsx]: Actually, I would prefer to do defensive programming here instead

```typescript
if (revealKeepAliveData.fdmNodeCache.current !== undefined) {
  return;
}

revealKeepAliveData.fdmNodeCache.current = new FdmNodeCache(..)
```

Comment (haakonflatval-cognite) [reply react-components/src/components/NodeCacheProvider/NodeCacheProvider.tsx]: None of these will do what we want :sweat: I only want to set `revealKeepAliveData.fdmNodeCache.current` to the new cache if `revealKeepAliveData` itself is present. That check is not done in either of the snippets

**Review by Savokr — APPROVED** [review]
Looks good and works nice!

**Review by Savokr — APPROVED** [review]
Nice!

**Pull request merged**